### ISSUE TITLE
bug(datasets): remove unsafe PyTorch load and save

### ISSUE THREAD

**Issue by deepyaman** [kedro-org/kedro-plugins#850]
CI is failing due to https://bandit.readthedocs.io/en/latest/plugins/b704_pytorch_load_save.html, a Bandit rule that was introduced earlier today in their 1.7.10 release.

`safetensors` is one possible option for addressing this, but I think somebody who is using TensorFlow with Kedro should ideally weigh in.

**Comment by deepyaman** [comment]
@gitgud5000 I saw you have another PR with a proposed improvement to the `TensorFlowModelDataset`; assuming that means you're using it, would you be open to taking a look at what makes sense to you here. :)

**Comment by DimedS** [comment]
Thanks for flagging this, @deepyaman. Perhaps we can skip the check for now (especially since this is still an experimental dataset) and take some time to find a more permanent solution. There's also a possibility that PyTorch may update its methods in the future?

**Comment by merelcht** [comment]
> Perhaps we can skip the check for now (especially since this is still an experimental dataset) and take some time to find a more permanent solution.

This is not an experimental dataset, but perhaps we should demote it and ignore the issue until it's resolved properly.

**Comment by ankatiyar** [comment]
> > Perhaps we can skip the check for now (especially since this is still an experimental dataset) and take some time to find a more permanent solution.
> 
> This is not an experimental dataset, but perhaps we should demote it and ignore the issue until it's resolved properly.

@merelcht The `bandit` check is failing in `kedro_datasets_experimental.pytorch.PytorchDataset`. I've added `#nosec` comments in #845 to unblock the CI but we can leave this issue open for the author of the dataset to take care of?

**Comment by merelcht** [comment]
Oh my bad, I thought it was the `TensorFlowModelDataset` that was the problem! 

Yeah let's ping the author. They might be able to give insight into whether this is a real issue or safe to ignore.

**Comment by gitgud5000** [comment]
@deepyaman I agree—this is just wrapping the interface for saving/loading in-house objects, not external or intended for distribution... 

I haven't tried the safetensors with TensorFlow yet, but with Keras > 3, their serialization forces you to use the built-in methods/API anyway.

**Comment by bpmeek** [comment]
I can try to look into this more tomorrow. I'm a little confused though because when I wrote this I was attempting to follow [PyTorch's recommended load and save method](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-state-dict-recommended) but perhaps something has changed.
